Bottom: c77f3b5d07dd66276c1091d7da0633650e4fa83f
Top:    3ca407b40bea233d2bd5f8d0bbb3707fbe8bddaf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-20 15:43:46 +0100

Refresh of hid-gtk-rework-view-flip-code

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 1f023e6..8a38100 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -122,6 +122,23 @@ ghid_zoom_view_fit (void)
                                  PCB->MaxHeight / gport->height));
 }
 
+static void
+ghid_flip_view (Coord center_x, Coord center_y, bool flip_x, bool flip_y)
+{
+  int widget_x, widget_y;
+
+  /* Work out where on the screen the flip point is */
+  ghid_pcb_to_event_coords (center_x, center_y, &widget_x, &widget_y);
+
+  ghid_flip_x = ghid_flip_x != flip_x;
+  ghid_flip_y = ghid_flip_y != flip_y;
+
+  /* Pan the board so the center location remains in the same place */
+  ghid_pan_view_abs (center_x, center_y, widget_x, widget_y);
+
+  ghid_invalidate_all ();
+}
+
 /* ------------------------------------------------------------ */
 
 static const char zoom_syntax[] =
